src/resource/user/user.service.ts

```diff
@@ -36,6 +38,24 @@ export class UserService {
     return this.prisma.user.findFirst({ where: { email: _email } });
   }
 
+  async findByResetPasswordToken(_resetToken: string) {
+    const user = await this.prisma.user.findFirst({
+      where: { passwordResetToken: _resetToken },
+    });
+    const { passwordResetTokenExpires } = user;
+    const expire_time = Date.parse(passwordResetTokenExpires);
```

Comment:
This feels like it belongs to `authService`?

Reply:
This expire time check has been moved to authService